carts/views.py

```diff
@@ -22,11 +23,32 @@ def post(self, request):
                 cart.quantity += 1
                 cart.save()
             else:
-                return JsonResponse({'message': 'QUANTITY_CHANGED'}, status=200)
+                return JsonResponse({'message': 'INVALID_QUANTITY'}, status=200)
 
             return JsonResponse({'message': 'CART_CREATED'}, status=201)
 
         except Cart.DoesNotExist:
             return JsonResponse({'message': 'CART_DOES_NOT_EXIT'}, status=400)
         except KeyError:
             return JsonResponse({'message': 'KEY_ERROR'}, status=400)
+
+    @author
+    def patch(self, request):
+        try:
+            data       = json.loads(request.body)
+            user       = request.user
+            product_id = data['product_id']
+            quantity   = data['quantity']
+
+            if quantity <= 0 or quantity >= 21:
+                return JsonResponse({'message': 'INVALID_QUANTITY'}, status=400)
+
+            if Cart.objects.filter(product_id=product_id, user_id=user).exists():
+                cart          = Cart.objects.get(user=user, product_id=product_id)
+                cart.quantity = data['quantity']
+                cart.save()
+                return JsonResponse({'message': 'QUANTITY_CHANGED'}, status=201)
+            return JsonResponse({'message': 'PRODUCT_DOES_NOT_EXIT'}, status=404)
```

Comment:
잘 작성해 주셨습니다-! 👍🏻 
다만 cart 데이터를 업데이트하는 엔드포인트이기 때문에 `PATCH :8000/carts/1`가 될 수 있도록 product_id가 아니라 cart_id 기준으로 로직을 수정해주세요-!

```suggestion
    def patch(self, request, cart_id):
        try:
            data       = json.loads(request.body)
            quantity   = data['quantity']

            if quantity <= 0 or quantity >= 21:
                return JsonResponse({'message': 'INVALID_QUANTITY'}, status=400)

            if Cart.objects.filter(id=cart_id, user_id=user).exists():
                cart = Cart.objects.get(id = cart_id, user=user)
                cart.quantity = data['quantity']
                cart.save()
                
                return JsonResponse({'message': 'QUANTITY_CHANGED'}, status=201)
                
            return JsonResponse({'message': 'PRODUCT_DOES_NOT_EXIT'}, status=404)
```